lib/appsignal/loaders/hanami.rb

```diff
@@ -23,9 +27,12 @@ def on_start
         )
         hanami_app_config.middleware.use(Appsignal::Rack::HanamiMiddleware)
 
+        return unless Gem::Version.new(Hanami::VERSION) < Gem::Version.new("2.2.0")
+
         ::Hanami::Action.prepend Appsignal::Loaders::HanamiLoader::HanamiIntegration
       end
 
+      # Legacy instrumentation to set the action name in Hanami apps older than Hanami 2.2
```

Comment:
Nit: I wonder if it would make sense to modify the legacy instrumentation so that it sets the `HANAMI_ACTION_INSTANCE` attribute, in the same way that Hanami 2.2 does -- that way, the legacy code path converges into the new one.

Reply:
I'm leaving out the suggested change. I'd rather not mess with the Hanami request env. Testing this is also a bit cumbersome / detached from the two implementations.